api/tasks.go

```diff
@@ -0,0 +1,567 @@
+	// Cron holds cron-like setting, e.g. once an hour at beginning of the hour "0 * * * *".
+	CreateTaskWithCron(ctx context.Context, name, flux, cron, orgID string) (*domain.Task, error)
+	// UpdateTask updates a task.
+	// It copies Description, Flux, Status, Offset and Every or Cron properties. Every and Cron are mutually exclusive.
+	// Every has higher priority.
+	UpdateTask(ctx context.Context, task *domain.Task) (*domain.Task, error)
+	// DeleteTask deletes a task.
+	DeleteTask(ctx context.Context, task *domain.Task) error
+	// DeleteTaskWithID deletes a task with taskID.
+	DeleteTaskWithID(ctx context.Context, taskID string) error
+	// FindMembers returns members of a task.
+	FindMembers(ctx context.Context, task *domain.Task) ([]domain.ResourceMember, error)
+	// FindMembersWithID returns members of a task with taskID.
+	FindMembersWithID(ctx context.Context, taskID string) ([]domain.ResourceMember, error)
+	// AddMember adds a member to a task.
+	AddMember(ctx context.Context, task *domain.Task, user *domain.User) (*domain.ResourceMember, error)
+	// AddMember adds a member with id memberID to a task with taskID.
+	AddMemberWithID(ctx context.Context, taskID, memberID string) (*domain.ResourceMember, error)
+	// RemoveMember removes a member from a task.
+	RemoveMember(ctx context.Context, task *domain.Task, user *domain.User) error
+	// RemoveMember removes a member with id memberID from a task with taskID.
+	RemoveMemberWithID(ctx context.Context, taskID, memberID string) error
+	// FindOwners returns owners of a task.
+	FindOwners(ctx context.Context, task *domain.Task) ([]domain.ResourceOwner, error)
+	// FindOwnersWithID returns owners of a task with taskID.
+	FindOwnersWithID(ctx context.Context, taskID string) ([]domain.ResourceOwner, error)
+	// AddOwner adds an owner to a task.
+	AddOwner(ctx context.Context, task *domain.Task, user *domain.User) (*domain.ResourceOwner, error)
+	// AddOwner adds an owner with id memberID to a task with taskID.
+	AddOwnerWithID(ctx context.Context, taskID, memberID string) (*domain.ResourceOwner, error)
+	// RemoveOwner removes an owner from a task.
+	RemoveOwner(ctx context.Context, task *domain.Task, user *domain.User) error
+	// RemoveOwner removes a member with id memberID from a task with taskID.
+	RemoveOwnerWithID(ctx context.Context, taskID, memberID string) error
+	// FindRuns retrieves a task runs according the filter. More fields can be applied. Filter can be nil.
+	FindRuns(ctx context.Context, task *domain.Task, filter *RunFilter) ([]domain.Run, error)
+	// FindRunsWithID retrieves runs of a task with taskID according the filter. More fields can be applied. Filter can be nil.
+	FindRunsWithID(ctx context.Context, taskID string, filter *RunFilter) ([]domain.Run, error)
+	// FindRunByID returns specific task run by taskID and runID
+	FindRunByID(ctx context.Context, taskID, runID string) (*domain.Run, error)
```

Comment:
Can be easily confused with `FindRunsWithID` ... suggestion: rename to `GetRunByID`

Reply:
Ok. For retrieving a single specific entity instance there are Get* and Get*ByID functions.